### ISSUE TITLE
`@enum.verify(enum.UNIQUE)` fails with "error: Argument of type "str" cannot be assigned to parameter"

### ISSUE THREAD

**Issue by ottumm** [microsoft/pyright#8079]
The following example succeeds with Pyright 1.1.365 and fails with 1.1.366 with the following error:

> Argument of type "str" cannot be assigned to parameter "checks" of type "EnumCheck" in function "__init__"
  "str" is incompatible with "EnumCheck"  (reportArgumentType)

Code sample in [pyright playground](https://pyright-play.net/?code=GYJw9gtgBApgdgV2gSwgBzCALlAqgOQEkBFXAUQBoozEIqA3GEZYATwChOABR5tgCgIlyASnYBjADYBDAM6yoAMTBh%2BNJCIBc7KLqgBBKAF4oAImmmdegELGzAI0tA)

```python
from enum import UNIQUE, Enum, verify


@verify(UNIQUE)
class Foo(Enum):
    A = "a"
    B = "b"

```

**Comment by erictraut** [comment]
This is a bug in the typeshed stubs. This previously worked because of an ambiguity in the typing spec. This ambiguity was recently clarified, and pyright 1.1.366 implements this change. 

I thought we had found and fixed all of these issues in the typeshed stubs a while ago, but apparently this one wasn't addressed.

The problem is specifically with the `EnumCheck` and `FlagBoundary` class definitions in `enum.pyi`. The enum members in these two classes are defined in a way that indicates they should be interpreted as non-member attributes rather than members. They should look like:

```python
    class EnumCheck(StrEnum):
        CONTINUOUS = ...
        NAMED_FLAGS = ...
        UNIQUE = ...
```

I've [created a PR](https://github.com/python/typeshed/pull/12100) in the typeshed project to fix this.

**Comment by ottumm** [comment]
Cool, thanks for tracking that down and opening the pr!

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.367.
